libcperciva/events/events_network.c

```diff
@@ -25,7 +25,7 @@
  * the value INT_MAX + 1 (in case every possible file descriptor is in use
  * and being polled for).
  */
-CTASSERT((nfds_t)(-1) <= (size_t)(-1));
+CTASSERT((nfds_t)(-1) <= (size_t)(SIZE_MAX));
```

Comment:
Thanks for your interest, @dorjoy03!

As far as I'm aware, `(size_t)-1` is identical to `(size_t)SIZE_MAX`; so this is a stylistic change.

In this case,

    CTASSERT((nfds_t)(-1) <= (size_t)(-1));

looks more clear to me: the only difference between the left-hand side and right-hand side is the type, so we immediately see that it's comparing the types.

Could you please elaborate on why you think `SIZE_MAX` is more appropriate here?

Reply:
My main motivation for this change here was that SIZE_MAX is more explicit as it's provided in the stdint.h header i.e., the maximum value a size_t type can have and by (size_t)(-1) we do mean exactly that but it is more implicit than SIZE_MAX. I guess for me (size_t)(-1) would make more sense if there was no explicit max value provided in any header. 

But I can revert this change here if the previous version is more clear to you. Let me know. I can do a force push together with other changes needed (if any).

Reply:
I think that (size_t)(-1) is actually more idiomatic in these cases, so unfortunately the first commit isn't necessary.  In system C programming, there's a lot of history behind `(int)(-1)` indicating a "failure" or "invalid value"; `(size_t)(-1)` is doing the same thing here.

For example, take a look at some of the results for
https://github.com/search?q=repo%3Afreebsd%2Ffreebsd-src+%22%28size_t%29-1%22&type=code
in the FreeBSD source code.  (As a side note, they seem to prefer `(size_t)-1`, i.e. without the parentheses around the `-1`, which isn't how I'd write it, but hey, it's not my code!)

As another example, consider the `mbrlen()` and `mbrtowc()` multibyte functions in C99.  They return `(size_t)(-1)` or even `(size_t)(-2)` to indicate that something funky is happening.

Other than the `CTASSERT`, we're always using `(size_t)(-1)` to indicate an invalid value; so I think it's more appropriate to keep them as such, rather than replacing them with `SIZE_MAX`.

Reply:
Got it. Makes sense. I will remove the first commit. Thanks!